service/application/src/main/java/com/solace/maas/ep/event/management/agent/service/EventBrokerResourceConfigToEventConverter.java

```diff
@@ -0,0 +1,82 @@
+package com.solace.maas.ep.event.management.agent.service;
+
+import com.solace.maas.ep.common.model.EventBrokerAuthenticationConfiguration;
+import com.solace.maas.ep.common.model.EventBrokerConnectionConfiguration;
+import com.solace.maas.ep.common.model.EventBrokerCredentialConfiguration;
+import com.solace.maas.ep.common.model.EventBrokerResourceConfiguration;
+import com.solace.maas.ep.event.management.agent.event.MessagingServiceEvent;
+import com.solace.maas.ep.event.management.agent.plugin.messagingService.event.AuthenticationDetailsEvent;
+import com.solace.maas.ep.event.management.agent.plugin.messagingService.event.AuthenticationOperationDetailsEvent;
+import com.solace.maas.ep.event.management.agent.plugin.messagingService.event.ConnectionDetailsEvent;
+import com.solace.maas.ep.event.management.agent.plugin.messagingService.event.CredentialDetailsEvent;
+import com.solace.maas.ep.event.management.agent.plugin.messagingService.event.EventProperty;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.Objects;
+
+@Service
+public class EventBrokerResourceConfigToEventConverter {
+    public MessagingServiceEvent mapToMessagingServiceEvent(EventBrokerResourceConfiguration eventBrokerResource) {
+        Objects.requireNonNull(eventBrokerResource);
+        MessagingServiceEvent serviceEvent = new MessagingServiceEvent();
+        serviceEvent.setMessagingServiceType(eventBrokerResource.getBrokerType());
+        serviceEvent.setName(eventBrokerResource.getName());
+        serviceEvent.setId(eventBrokerResource.getId());
+        serviceEvent.setMessagingServiceType(eventBrokerResource.getResourceConfigurationType().name());
+
+        ConnectionDetailsEvent connectionDetailsEvent = new ConnectionDetailsEvent();
+        EventBrokerConnectionConfiguration connectionDetail = eventBrokerResource.getConnections().get(0);
+
+        connectionDetailsEvent.setName(eventBrokerResource.getName());
+        connectionDetailsEvent.setUrl(connectionDetail.getUrl());
+        connectionDetailsEvent.setMessagingServiceId(eventBrokerResource.getId());
+        connectionDetailsEvent.setProperties(List.of(
+                EventProperty.builder()
+                        .name("sempPageSize")
```

Comment:
I think it might help readability if this class or method name contains `SolaceBroker` instead of the generic names.

Reply:
Done